README.md

```diff
@@ -15,3 +15,20 @@ part of the array each recursive call considers.
 Analyse the time complexity of your implementation and give a $\Theta$ bound for
 its worst-case runtime. Add your answer, including your reasoning, to this
 markdown file.
+
+Steps of my Function:
+1. Determine how large my function is in terms of powers of two. Constant Time operation
+2. Increment $x$ by powers of two $log_2(n)$ times, running step 3 for each power of x
+3. Step through the array by x to select segemnts to run step 4 on. $log_x(n)$
```

Comment:
Is this another $\log n$ step? How many times will it run? It wouldn't be $\log n$ if the step size is $\log n$.

Reply:
Hmm... It is taking steps of logn until it is larger than n. Would that be n/log(n) then? Which would then result in log(n)*n/log(n) = n for those loops, which overall would be n^3, or something.

Also, is there a good way to do mathematical notation in these comments?

Reply:
I think I see. The purpose of the loops is essentially to break the array down into sections. Each element in these sections gets processed $log_2(n)$ times, so between the two loops, that would be the complexity. That brings the overall complexity to $n^2log(n)$. That makes more sense, as the additional factor of $n$ in the code is just from making it in-place. If I dropped the constraint of the solution being in-place by using a stack of variable memory, it would then have a complexity of $nlog(n)$ which matches the theoretical complexity of mergesort.

Reply:
Sounds good!